### DISCUSSION TITLE
Appropriate version lock for openai dependency

### DISCUSSION THREAD

**Discussion by arksenu** [AntonOsika/gpt-engineer#907] [General]
This project already I recall was using 0.28 for some reason now it's downgraded again? Or what am I not getting this is fucking the api syntax up

**Comment by ErikBjare** [comment]
I don't know what you mean.

`pyproject.toml` contains `openai = "0.28"`

**Comment by ATheorell** [comment]
The openai version is a bit of a pickle. Too low versions get us in trouble with model availability. Versions from 1.0.0 and upwards has a new syntax (as you point out). I tried to migrate to the new syntax, but it led to conflicts with other dependencies. Will have a new shot at migrating soon, so that we stay up to date for the softwares using gpt-engineer as a dependency.
